ethereum/core/build.gradle

```diff
@@ -51,6 +52,7 @@ dependencies {
 
   implementation 'com.fasterxml.jackson.core:jackson-databind'
   implementation 'com.google.guava:guava'
+  implementation 'com.github.ben-manes.caffeine:caffeine:3.0.4'
```

Comment:
Some of the [limitations in how weighted caches](https://groups.google.com/g/guava-discuss/c/P4f13-SH1Rc) are handled made this very hard to test. Upgrade was easy, and interfaces are very similar. Suggest using it elsewhere also.